src/esp/gfx_batch/Renderer.h

```diff
@@ -703,6 +714,16 @@ class Renderer {
    */
   void draw(Magnum::GL::AbstractFramebuffer& framebuffer);
 
+  /**
+   * @brief Unprojects depth onto the provided image.
+   * Produces an image that contains the distance of each pixels, in meters.
+   *
+   * @param sceneId Scene ID, expected to be less than @ref sceneCount()
+   * @param view View on raw depth of scene corresponding to the specified
+   * sceneId.
+   */
+  void unprojectDepth(int sceneId, const Magnum::MutableImageView2D& view);
```

Comment:
By user, you mean `BatchReplayRenderer` or replay renderer API user?

I wouldn't mind moving this elsewhere, but I'd make sure that the behavior is the same for the classic and batch replay renderer, which is to return an unprojected depth image when calling `render()`.

Reply:
Users, as in currently the `BatchReplayRenderer`, yes. 

`BatchReplayRenderer::render()` should make sure that both the classic and the batch renderer return an unprojected depth image, yes, but this functionality is orthogonal to the `gfx_patch::Renderer` class. There's nothing "batch" about it.

Tthere isn't anything this API would use from the batch renderer, except for the unprojection parameters which is literally two floats -- the API could even be `const`. Having it as a member function here adds no value, doesn't really help with anything, only introduces a lot of implicit constraints for no reason. By having it here, you're making it harder to use a GPU-based unprojection instead, or to do it in a single pass across in multiple environments. 

Remove, and expose just a `Vector2 cameraDepthUnprojection(Magnum::UnsignedInt sceneId) const` getter, and in the `BatchReplayRenderer` you'd then simply call the underlying API directly. So instead of

```cpp
renderer_->unprojectDepth(envIndex, depthBufferView);
```

you'd have something like this. Not any more complicated than the original and you remove a lot of unnecessary coupling.

```cpp
gfx_batch::unprojectDepth(renderer->cameraDepthUnprojection(envIndex), depthBufferView));
```